panic occured when wrap go-playground./validator's errors

from v4.0.0


```go
package main

import (
	"fmt"

	"github.com/srvc/fail"
	validator "gopkg.in/go-playground/validator.v9"
)

func main() {
	err := validator.ValidationErrors{}
	fmt.Println(fail.Wrap(err))
}
```


```
panic: runtime error: comparing uncomparable type validator.ValidationErrors

goroutine 1 [running]:
github.com/srvc/fail.extractPkgError(0x1be6a0, 0x40cfe0, 0x450230, 0x277c00)
	/go/src/github.com/srvc/fail/pkgerrors.go:82 +0x960
github.com/srvc/fail.convertPkgError(0x1be6a0, 0x40cfe0, 0x1901c0, 0x9f442304)
	/go/src/github.com/srvc/fail/pkgerrors.go:22 +0x40
github.com/srvc/fail.Unwrap(0x1be6a0, 0x40cfe0, 0x15e901, 0x7622)
	/go/src/github.com/srvc/fail/error.go:113 +0x260
github.com/srvc/fail.Wrap(0x1be6a0, 0x40cfe0, 0x0, 0x0, 0x0, 0x4423a0, 0x0, 0x0)
	/go/src/github.com/srvc/fail/error.go:85 +0x40
main.main()
	/go/src/29f35e98-5c00-4964-a695-ff39d667e9e2/main.go:12 +0xa0
```

--- a/pkgerrors.go
+++ b/pkgerrors.go
@@ -86,7 +86,7 @@ func extractPkgError(err error) *pkgError {
 			if rootErr == err {
 				return nil
 			}
-		} else if reflect.DeepEqual(rootErr, err) {
+		} else {
 			return nil
 		}
 	}

--- a/pkgerrors.go
+++ b/pkgerrors.go
@@ -1,6 +1,7 @@
 package fail
 
 import (
+	"reflect"
 	"strings"
 
 	pkgerrors "github.com/pkg/errors"
@@ -79,8 +80,15 @@ func extractPkgError(err error) *pkgError {
 		break
 	}
 
-	if len(stackTraces) == 0 && rootErr == err {
-		return nil
+	if len(stackTraces) == 0 {
+		ret, et := reflect.TypeOf(rootErr), reflect.TypeOf(err)
+		if ret != nil && et != nil && ret.Comparable() && et.Comparable() {
+			if rootErr == err {
+				return nil
+			}
+		} else if reflect.DeepEqual(rootErr, err) {
+			return nil
+		}
 	}
 
 	// Extract annotated messages by removing the trailing message.

--- a/pkgerrors_test.go
+++ b/pkgerrors_test.go
@@ -2,6 +2,7 @@ package fail
 
 import (
 	"errors"
+	"strings"
 	"testing"
 
 	pkgerrors "github.com/pkg/errors"
@@ -17,6 +18,13 @@ func TestExtractPkgError(t *testing.T) {
 
 	t.Run("not a pkg/errors", func(t *testing.T) {
 		pkgErr := extractPkgError(errors.New("error"))
+		assert.Nil(t, pkgErr)
+	})
+
+	t.Run("slice error", func(t *testing.T) {
+		err := errorSlice{errors.New("error")}
+
+		pkgErr := extractPkgError(err)
 		assert.Nil(t, pkgErr)
 	})
 
@@ -83,6 +91,16 @@ func TestExtractPkgError(t *testing.T) {
 			assert.Equal(t, err0, pkgErr.Err)
 			assert.NotEmpty(t, pkgErr.StackTrace)
 			assert.Equal(t, "pkgErrorsWrap", pkgErr.StackTrace[0].Func)
+		})
+
+		t.Run("with slice error", func(t *testing.T) {
+			err0 := errorSlice{errors.New("error")}
+			err1 := pkgErrorsWrap(err0, "message")
+
+			pkgErr := extractPkgError(err1)
+			assert.NotNil(t, pkgErr)
+			assert.Equal(t, err0, pkgErr.Err)
+			assert.NotEmpty(t, pkgErr.StackTrace)
 		})
 	})
 
@@ -155,4 +173,14 @@ func pkgErrorsNew(msg string) error {
 
 func pkgErrorsWrap(err error, msg string) error {
 	return pkgerrors.Wrap(err, msg)
+}
+
+type errorSlice []error
+
+func (s errorSlice) Error() string {
+	msg := make([]string, len(s))
+	for i, e := range s {
+		msg[i] = e.Error()
+	}
+	return strings.Join(msg, ": ")
 }

